doc/whats_new/v1.1.rst

```diff
@@ -244,6 +244,9 @@ Changelog
   `decision_function` instead of returning the multilabel indicator matrix.
   :pr:`19869` by :user:`Guillaume Lemaitre <glemaitre>`.
 
+- |Fix| :class:`impute.SimpleImputer` now uses the dtype seen in `fit` for
+  `transform`. :pr:`22063` by `Thomas Fan`_.
+
```

Comment:
This should be move to `doc/whats_new/v1.2`

Reply:
Actually in 1.1.2. I assume that we will do another bug fix release at some point.

Reply:
This is likely too big of a change for `1.1.2`. Currently fitting on `float64` and transforming a `float32` would return `float32`:

```python
import numpy as np
from sklearn.impute import SimpleImputer
imp_mean = SimpleImputer(missing_values=np.nan, strategy='mean')
X = np.asarray([[7, 2, 3], [4, np.nan, 6], [10, 5, 9]], dtype=np.float64)
imp_mean.fit(X)

X_test = np.asarray([[np.nan, 2, 3], [4, np.nan, 6], [10, np.nan, 9]], dtype=np.float32)
X_trans = imp_mean.transform(X_test)

print(X_trans.dtype)
# float32
```

Looking at this again, I think it's better to error when fitting on an object dtype, but transforming on a non-object.